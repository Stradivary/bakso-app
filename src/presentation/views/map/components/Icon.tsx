import L from "leaflet";
import markerPerson from "@/shared/assets/cust.svg";
import markerBakso from "@/shared/assets/seller.svg";

const iconPerson = new L.Icon({
  iconUrl: markerPerson,
  iconSize: new L.Point(32, 49),
});

const iconBakso = new L.Icon({
  iconUrl: markerBakso,
  iconSize: new L.Point(32, 49),
});

export { iconPerson, iconBakso };
