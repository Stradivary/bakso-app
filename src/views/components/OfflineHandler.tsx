import { Modal, Title } from "@mantine/core";
import { useNetwork } from "@mantine/hooks";

export function OfflineHandler() {
  const { online } = useNetwork();
  return (
    <Modal
      title={
        <Title order={5} style={{ margin: 0 }}>
          Duh, Koneksi Internet Terputus!
        </Title>
      }
      opened={!online}
      onClose={() => {}}
      withCloseButton={false}
    >
      Sepertinya koneksi internet Anda terputus. Mohon periksa kembali koneksi
      internet Anda.
    </Modal>
  );
}
