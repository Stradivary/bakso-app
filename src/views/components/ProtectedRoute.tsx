import { Box, LoadingOverlay } from "@mantine/core";
import { Navigate } from "react-router-dom";
import { useAuth } from "@/shared/contexts/authProvider";

export const ProtectedRoute: React.FC<{ children: JSX.Element }> = ({
  children,
}) => {
  const { isAuthenticated, isLoading } = useAuth();

  if (isLoading) {
    return (
      <Box h="100dvh">
        <LoadingOverlay visible />
      </Box>
    );
  }

  if (!isAuthenticated) {
    return <Navigate to="/login" replace />;
  }

  return children;
};
