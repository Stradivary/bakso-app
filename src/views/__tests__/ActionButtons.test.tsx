import { render, screen } from "@testing-library/react";
import { describe, expect, it, vi } from "vitest";
import { ActionButtons } from "../components/ActionButtons";
import { MantineProvider } from "@mantine/core";
import { Notification } from "@/models/Notification.types";

describe("ActionButtons", () => {
  const defaultProps = {
    onRecenter: vi.fn(),
    onExit: vi.fn(),
    role: "customer",
    name: "John Doe",
  };

  it("renders correctly with customer role", () => {
    render(
      <MantineProvider>
        <ActionButtons {...defaultProps} />
      </MantineProvider>,
    );

    expect(screen.getByText("Customer")).toBeDefined();
    expect(screen.getByText("John Doe")).toBeDefined();
    expect(screen.queryByTestId("notification-center")).toBeNull();
  });

  it("renders correctly with seller role", () => {
    render(
      <MantineProvider>
        <ActionButtons {...defaultProps} role="seller" />
      </MantineProvider>,
    );

    expect(screen.getByText("Tukang Bakso")).toBeDefined();
    expect(screen.getByTestId("notification-center")).toBeDefined();
  });

  it("calls onRecenter when recenter button is clicked", () => {
    const { container } = render(
      <MantineProvider>
        <ActionButtons {...defaultProps} />
      </MantineProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it("calls onExit when exit button is clicked", () => {
    const { container } = render(
      <MantineProvider>
        <ActionButtons {...defaultProps} />
      </MantineProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it("renders notifications when provided and role is seller", () => {
    const notifications = [
      {
        id: "1",
        buyer_id: "Test notification",
        created_at: "",
        expiry_at: "0",
        is_read: false,
        seller_id: "",
        buyer_name: "",
      },
    ] as Notification[];

    const { container } = render(
      <MantineProvider>
        <ActionButtons
          {...defaultProps}
          role="seller"
          notifications={notifications}
        />
      </MantineProvider>,
    );

    expect(container).toMatchSnapshot();
  });

  it("applies correct badge colors based on role", () => {
    const { container, rerender } = render(
      <MantineProvider>
        <ActionButtons {...defaultProps} />
      </MantineProvider>,
    );

    expect(container).toMatchSnapshot();

    rerender(
      <MantineProvider>
        <ActionButtons {...defaultProps} role="seller" />
      </MantineProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
