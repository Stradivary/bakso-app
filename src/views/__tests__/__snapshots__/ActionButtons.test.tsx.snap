// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionButtons > applies correct badge colors based on role 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="z-index: 50; position: absolute; top: 0px; right: 0px; display: flex; flex-direction: column; align-items: flex-end; padding: 8px;"
  >
    <div
      class="m_8bffd616 mantine-Flex-root __m__-r24"
      style="flex-direction: row;"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: calc(0.375rem * var(--mantine-scale)); --group-align: start; --group-justify: end; --group-wrap: wrap; user-select: none; pointer-events: painted;"
      >
        <div
          class="m_347db0ec mantine-Badge-root"
          style="--badge-bg: var(--mantine-color-blue-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            Customer
          </span>
        </div>
        <div
          class="m_347db0ec mantine-Badge-root"
          data-variant="filled"
          style="--badge-bg: var(--mantine-color-teal-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            John Doe
          </span>
        </div>
      </div>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-md); --stack-align: end; --stack-justify: flex-start; flex-grow: 1;"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-blue-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-locate-fixed"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="2"
                  x2="5"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="19"
                  x2="22"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="5"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="19"
                  y2="22"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="7"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-red-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-x"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActionButtons > applies correct badge colors based on role 2`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="z-index: 50; position: absolute; top: 0px; right: 0px; display: flex; flex-direction: column; align-items: flex-end; padding: 8px;"
  >
    <div
      class="m_8bffd616 mantine-Flex-root __m__-r24"
      style="flex-direction: row;"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: calc(0.375rem * var(--mantine-scale)); --group-align: start; --group-justify: end; --group-wrap: wrap; user-select: none; pointer-events: painted;"
      >
        <div
          class="m_347db0ec mantine-Badge-root"
          style="--badge-bg: var(--mantine-color-green-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            Tukang Bakso
          </span>
        </div>
        <div
          class="m_347db0ec mantine-Badge-root"
          data-variant="filled"
          style="--badge-bg: var(--mantine-color-teal-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            John Doe
          </span>
        </div>
      </div>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-md); --stack-align: end; --stack-justify: flex-start; flex-grow: 1;"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-blue-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-locate-fixed"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="2"
                  x2="5"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="19"
                  x2="22"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="5"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="19"
                  y2="22"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="7"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-red-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-x"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </span>
        </button>
        <div
          aria-controls=""
          aria-expanded="false"
          aria-haspopup="dialog"
          class="m_e5262200 mantine-Indicator-root"
          id=""
          style="--indicator-size: calc(0.75rem * var(--mantine-scale)); --indicator-top: 0rem; --indicator-right: 0rem; --indicator-translate-x: 50%; --indicator-translate-y: -50%;"
        >
          <div
            class="m_760d1fb1 mantine-Indicator-indicator"
            data-processing="true"
          />
          <button
            class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
            data-size="compact-lg"
            data-testid="notification-center"
            data-variant="white"
            style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-blue-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
            type="button"
          >
            <span
              class="m_80f1301b mantine-Button-inner"
            >
              <span
                class="m_811560b9 mantine-Button-label"
              >
                <svg
                  class="lucide lucide-bell"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 8a6 6 0 0 1 12 0c0 7 3 9 3 9H3s3-2 3-9"
                  />
                  <path
                    d="M10.3 21a1.94 1.94 0 0 0 3.4 0"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActionButtons > calls onExit when exit button is clicked 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="z-index: 50; position: absolute; top: 0px; right: 0px; display: flex; flex-direction: column; align-items: flex-end; padding: 8px;"
  >
    <div
      class="m_8bffd616 mantine-Flex-root __m__-r18"
      style="flex-direction: row;"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: calc(0.375rem * var(--mantine-scale)); --group-align: start; --group-justify: end; --group-wrap: wrap; user-select: none; pointer-events: painted;"
      >
        <div
          class="m_347db0ec mantine-Badge-root"
          style="--badge-bg: var(--mantine-color-blue-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            Customer
          </span>
        </div>
        <div
          class="m_347db0ec mantine-Badge-root"
          data-variant="filled"
          style="--badge-bg: var(--mantine-color-teal-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            John Doe
          </span>
        </div>
      </div>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-md); --stack-align: end; --stack-justify: flex-start; flex-grow: 1;"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-blue-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-locate-fixed"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="2"
                  x2="5"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="19"
                  x2="22"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="5"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="19"
                  y2="22"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="7"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-red-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-x"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActionButtons > calls onRecenter when recenter button is clicked 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="z-index: 50; position: absolute; top: 0px; right: 0px; display: flex; flex-direction: column; align-items: flex-end; padding: 8px;"
  >
    <div
      class="m_8bffd616 mantine-Flex-root __m__-rt"
      style="flex-direction: row;"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: calc(0.375rem * var(--mantine-scale)); --group-align: start; --group-justify: end; --group-wrap: wrap; user-select: none; pointer-events: painted;"
      >
        <div
          class="m_347db0ec mantine-Badge-root"
          style="--badge-bg: var(--mantine-color-blue-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            Customer
          </span>
        </div>
        <div
          class="m_347db0ec mantine-Badge-root"
          data-variant="filled"
          style="--badge-bg: var(--mantine-color-teal-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            John Doe
          </span>
        </div>
      </div>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-md); --stack-align: end; --stack-justify: flex-start; flex-grow: 1;"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-blue-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-locate-fixed"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="2"
                  x2="5"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="19"
                  x2="22"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="5"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="19"
                  y2="22"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="7"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-red-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-x"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`ActionButtons > renders notifications when provided and role is seller 1`] = `
<div>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class=""
    style="z-index: 50; position: absolute; top: 0px; right: 0px; display: flex; flex-direction: column; align-items: flex-end; padding: 8px;"
  >
    <div
      class="m_8bffd616 mantine-Flex-root __m__-r1j"
      style="flex-direction: row;"
    >
      <div
        class="m_4081bf90 mantine-Group-root"
        style="--group-gap: calc(0.375rem * var(--mantine-scale)); --group-align: start; --group-justify: end; --group-wrap: wrap; user-select: none; pointer-events: painted;"
      >
        <div
          class="m_347db0ec mantine-Badge-root"
          style="--badge-bg: var(--mantine-color-green-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            Tukang Bakso
          </span>
        </div>
        <div
          class="m_347db0ec mantine-Badge-root"
          data-variant="filled"
          style="--badge-bg: var(--mantine-color-teal-filled); --badge-color: var(--mantine-color-white); --badge-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
        >
          <span
            class="m_5add502a mantine-Badge-label"
          >
            John Doe
          </span>
        </div>
      </div>
      <div
        class="m_6d731127 mantine-Stack-root"
        style="--stack-gap: var(--mantine-spacing-md); --stack-align: end; --stack-justify: flex-start; flex-grow: 1;"
      >
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-blue-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-locate-fixed"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <line
                  x1="2"
                  x2="5"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="19"
                  x2="22"
                  y1="12"
                  y2="12"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="2"
                  y2="5"
                />
                <line
                  x1="12"
                  x2="12"
                  y1="19"
                  y2="22"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="7"
                />
                <circle
                  cx="12"
                  cy="12"
                  r="3"
                />
              </svg>
            </span>
          </span>
        </button>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-size="compact-lg"
          data-variant="white"
          style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-red-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
          type="button"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              <svg
                class="lucide lucide-x"
                fill="none"
                height="20"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="20"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 6 6 18"
                />
                <path
                  d="m6 6 12 12"
                />
              </svg>
            </span>
          </span>
        </button>
        <div
          aria-controls=""
          aria-expanded="false"
          aria-haspopup="dialog"
          class="m_e5262200 mantine-Indicator-root"
          id=""
          style="--indicator-size: calc(0.75rem * var(--mantine-scale)); --indicator-top: 0rem; --indicator-right: 0rem; --indicator-translate-x: 50%; --indicator-translate-y: -50%;"
        >
          <div
            class="m_760d1fb1 mantine-Indicator-indicator"
            data-processing="true"
          />
          <button
            class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
            data-size="compact-lg"
            data-testid="notification-center"
            data-variant="white"
            style="--button-height: var(--button-height-compact-lg); --button-padding-x: var(--button-padding-x-compact-lg); --button-fz: var(--mantine-font-size-lg); --button-bg: var(--mantine-color-white); --button-hover: rgba(252, 252, 252, 1); --button-color: var(--mantine-color-blue-filled); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent;"
            type="button"
          >
            <span
              class="m_80f1301b mantine-Button-inner"
            >
              <span
                class="m_811560b9 mantine-Button-label"
              >
                <svg
                  class="lucide lucide-bell"
                  fill="none"
                  height="24"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M6 8a6 6 0 0 1 12 0c0 7 3 9 3 9H3s3-2 3-9"
                  />
                  <path
                    d="M10.3 21a1.94 1.94 0 0 0 3.4 0"
                  />
                </svg>
              </span>
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
